pretty_gpx/common/data/overpass_processing.py

```diff
@@ -143,8 +143,8 @@ def get_polygons_from_relation(relation: Relation) -> list[ShapelyPolygon]:
         #Tolerance is 2m near the point (converted to lon/lat)
```

Comment:
hi @JulesL2 

I've just added a template in PR https://github.com/ThomasParistech/pretty-gpx/pull/46 :)

You should reuse it for the city mode. Resulting code should be quite compact since you don't have to worry about low/high res

Take inspiration from pretty_gpx/ui/pages/mountain/page.py

Reply:
@ThomasParistech 

Thank you, I used the template, I have still the same issue as before that I am trying to fix.
On large maps ie with a lot of things to draw, the ui.update() task seems to crash/I have a connection lost in the gui.

I have tried to mess with async functions and also matplotlib backend (sometimes I had tktinker errors) and I have not managed to find a good solution.

I let you try (I have included in this PR/branch the Paris Versailles run that is ok only if the backend is agg or svg. And you can also try with the paris-marathon example which always crash.

Otherwise the current code works perfectly for runs with not too much data (even Lyon Marathon is ok)

Reply:
Thanks, I'll have a look

Reply:
Working on it